### PR TITLE
sample(): added sample oauth2-client

### PR DESCRIPTION
**Pull request by Tony133**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
This PR, added a simple example of how to implement oauth2-client. PR for documentation: https://github.com/nestjs/docs.nestjs.com/pull/2245

### CODE REVIEW

**Comment by Tony133** [comment]
I thought about creating this example as I saw that it is a missing topic in the NestJS documentation.
It is true that OAuth2 is a somewhat complex topic to understand at times for those who have never used it, however I think a section should be added in the documentation, even if not that extensive.

What do you think about it? if this sample is approved, I start preparing a pull for the oauth2-client documentation.

Let me know what you think.

We would then have to make another sample unrelated to this one for oauth2-server.

He hoped it would help.

Tony

P.s. NestJS ♥️ 🙌

File: sample/31-oauth2-client/src/auth/strategies/google.strategy.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    async validate(accessToken: string, refreshToken: string, profile: any, done: VerifyCallback): Promise<any> {
+        const { name, emails, photos } = profile
+        const user = {
+            email: emails[0].value,
+            firstName: name.givenName,
+            lastName: name.familyName,
+            picture: photos[0].value,
+            accessToken
+        }
+        done(null, user);
```
Comment (jmcdo29) [inline sample/31-oauth2-client/src/auth/strategies/google.strategy.ts:25]: If we don't use `done` here, and just `return user` do we get the same effect? As we're also using promises, I'd rather not introduce callbacks as well.

Comment (Tony133) [reply sample/31-oauth2-client/src/auth/strategies/google.strategy.ts:25]: I tried now if I `return user` it works the same

Comment (jmcdo29) [reply sample/31-oauth2-client/src/auth/strategies/google.strategy.ts:25]: I would, yeah. Just to keep consistency

Comment (Tony133) [reply sample/31-oauth2-client/src/auth/strategies/google.strategy.ts]: Done

Comment (Mutondi) [reply sample/31-oauth2-client/src/auth/strategies/google.strategy.ts]: Please approve this comrades

**Comment by kamilmysliwiec** [comment]
I don't think we'll be adding more samples to this repository unless absolutely necessary. Thanks for your contribution anyway!

**Pull request closed without merging**